frontend/hooks/data-fetching/course.ts

```diff
@@ -30,6 +31,9 @@ import {
 export const useCourse = (courseId: number, initialCourse: Course) =>
     useResource(`/courses/${courseId}/`, initialCourse);
 
+export const useTags = (courseId: number, initialTags: Tag[]) =>
```

Comment:
Saw you attempted to use the resource hooks but ended up writing most of the API calls yourself below. 

You are definitely on the right track, but was using the wrong type of hook. What you're looking for is the useResourceList hook (https://github.com/pennlabs/frontend-core/blob/master/packages/rest-hooks/src/useResourceList.ts source code here). Basically, the hook provides you with tag data (which replaces getTags) and you can delete tags using the `mutate()` function by feeding in the id. Note that even though the routes for deleting a specific tag and getting all tags is slightly different, the 2nd parameter in the resource list hook allows you to specify the route of a specific tag item. A good reference for how to use this hook is useQuestions (which is also in this file). Unfortunately you still need createTag, similar to how we still have createQuestion. 

This should save some boilerplate with useEffects and keeping state in the components.